filament/src/RenderPass.cpp

```diff
@@ -496,15 +486,12 @@ void RenderPass::generateCommandsImpl(uint32_t,
                 cmdDepth.primitive.rasterState.culling = rs.culling;
                 *curr = cmdDepth;
 
-                // If we are drawing depth+draw we don't want to put commands using
-                // alpha testing (indicated by the alpha to coverage flag) or blending in the
-                // depth prepass. What we do want is put those commands in the shadow map
-                // (when only drawDepth is true).
-                // Also, depth-write could be disabled by the material,
-                // in this case undo the command.
-                bool issueDepth =
-                        (rs.depthWrite & !(colorPass & (rs.alphaToCoverage | rs.hasBlending())))
-                        | writeDepthForShadows;
+                // FIXME: should writeDepthForShadowCasters take precedence over rs.depthWrite?
+                bool issueDepth = (rs.depthWrite
```

Comment:
I'll trust you here…

Reply:
This is basically the same test than before, the differences are:
- it doesn't depend on "colorPass" anymore
- instead each of `alphaToCoverage` and `hasBlending` depend on a new specific flag (i.e. they can be controlled separately)